src/bundle.c

```diff
@@ -1058,13 +1058,190 @@ static struct list *get_dir_files_sorted(char *path)
 	return list_sort(files, lex_sort);
 }
 
+struct alias_lookup {
+	char *alias;
+	struct list *bundles;
+};
+
+/* Deep free on alias_lookup structs
+ */
+static void free_alias_lookup(void *lookup)
+{
+	struct alias_lookup *l = (struct alias_lookup *)lookup;
+	free(l->alias);
+	list_free_list_and_data(l->bundles, free);
+	free(l);
+}
+
+/* Get bundles for a given alias but if the alias is
+ * not actually an alias, return a list with just the
+ * bundle name
+ */
+static struct list *get_alias_bundles(struct list *alias_definitions, char *alias)
+{
+	struct list *bundles = NULL;
+	struct list *iter = alias_definitions;
+
+	while (iter) {
+		struct alias_lookup *lookup = (struct alias_lookup *)iter->data;
+		iter = iter->next;
+		if (strcmp(lookup->alias, alias) == 0) {
+			bundles = list_deep_clone_strs(lookup->bundles);
+			break;
+		}
+	}
+	if (!bundles) {
+		bundles = list_prepend_data(bundles, strdup_or_die(alias));
+	}
+
+	return bundles;
+}
+
+/* Parse an alias definition file for a list of bundles
+ * it specifies
+ *
+ * Example file:
+ * alias1	b1	b2
+ * alias2	b3	b4
+ */
+static struct list *parse_alias_file(char *fullpath)
+{
+	FILE *afile = NULL;
+	struct list *alias_content = NULL;
+	char *line = NULL;
+	size_t len = 0;
+	ssize_t read = -1;
+
+	afile = fopen(fullpath, "r");
+	if (!afile) {
+		return NULL;
+	}
+
+	while ((read = getline(&line, &len, afile)) != -1) {
+		struct alias_lookup *l = NULL;
+		char *c = NULL;
+
+		/* get the alias but ignore lines without bundles */
+		c = strchr(line, '\t');
```

Comment:
Size considerations of the file was pretty much the only reason for selecting this. I don't particularly see a reason to avoid more delimiters other than fewer options is safer than more for parsing.